src/main/java/de/tum/in/www1/artemis/domain/notification/SingleUserNotificationFactory.java

```diff
@@ -196,4 +201,95 @@ public static SingleUserNotification createNotification(TutorialGroup tutorialGr
+                String text = "You have been added to a new group chat by " + responsibleForAction.getName() + " in course (" + groupChat.getCourse().getTitle() + ").";
+                notification = new SingleUserNotification(user, title, text);
+                notification.setTransientAndStringTarget(createConversationCreationTarget(groupChat, groupChat.getCourse().getId()));
+                notification.setAuthor(responsibleForAction);
+            }
+            case CONVERSATION_ADD_USER_CHANNEL -> {
+                var channel = (Channel) conversation;
+                String text = "You have been added to channel (" + channel.getName() + ") by " + responsibleForAction.getName() + " in course (" + channel.getCourse().getTitle()
+                        + ").";
+                notification = new SingleUserNotification(user, title, text);
+                notification.setTransientAndStringTarget(createConversationCreationTarget(channel, channel.getCourse().getId()));
+                notification.setAuthor(responsibleForAction);
+            }
+            case CONVERSATION_REMOVE_USER_CHANNEL -> {
+                var channel = (Channel) conversation;
+                String text = "You have been removed from channel (" + channel.getName() + ") by " + responsibleForAction.getName() + " in course ("
+                        + channel.getCourse().getTitle() + ").";
+                notification = new SingleUserNotification(user, title, text);
+                notification.setTransientAndStringTarget(createConversationDeletionTarget(channel, channel.getCourse().getId()));
+                notification.setAuthor(responsibleForAction);
+            }
+            case CONVERSATION_REMOVE_USER_GROUP_CHAT -> {
+                var groupChat = (GroupChat) conversation;
+                String text = "You have been removed from group chat by " + responsibleForAction.getName() + " in course (" + groupChat.getCourse().getTitle() + ").";
+                notification = new SingleUserNotification(user, title, text);
+                notification.setTransientAndStringTarget(createConversationDeletionTarget(groupChat, groupChat.getCourse().getId()));
+                notification.setAuthor(responsibleForAction);
+            }
+            case CONVERSATION_DELETE_CHANNEL -> {
+                var channel = (Channel) conversation;
+                String text = channel.getName() + " channel has been deleted by " + responsibleForAction.getName() + " in course (" + channel.getCourse().getTitle() + ").";
+                notification = new SingleUserNotification(user, title, text);
+                notification.setTransientAndStringTarget(createConversationDeletionTarget(channel, channel.getCourse().getId()));
+                notification.setAuthor(responsibleForAction);
+            }
+            default -> throw new UnsupportedOperationException("Unsupported NotificationType: " + notificationType);
+        }
+        return notification;
+    }
```

Comment:
This extracts some statements and utilises the switch case syntax to its full extent.
```suggestion
        SingleUserNotification notification = switch (notificationType) {
            case CONVERSATION_CREATE_ONE_TO_ONE_CHAT -> {
                // text is null because the notification is not shown
                return new SingleUserNotification(user, title, null);
            }
            case CONVERSATION_CREATE_GROUP_CHAT, CONVERSATION_ADD_USER_GROUP_CHAT -> {
                var groupChat = (GroupChat) conversation;
                String text = "You have been added to a new group chat by " + responsibleForAction.getName() + " in course (" + groupChat.getCourse().getTitle() + ").";
                return new SingleUserNotification(user, title, text);
            }
            case CONVERSATION_ADD_USER_CHANNEL -> {
                var channel = (Channel) conversation;
                String text = "You have been added to channel (" + channel.getName() + ") by " + responsibleForAction.getName() + " in course (" + channel.getCourse().getTitle()
                        + ").";
                return new SingleUserNotification(user, title, text);
            }
            case CONVERSATION_REMOVE_USER_CHANNEL -> {
                var channel = (Channel) conversation;
                String text = "You have been removed from channel (" + channel.getName() + ") by " + responsibleForAction.getName() + " in course ("
                        + channel.getCourse().getTitle() + ").";
                return new SingleUserNotification(user, title, text);
            }
            case CONVERSATION_REMOVE_USER_GROUP_CHAT -> {
                var groupChat = (GroupChat) conversation;
                String text = "You have been removed from group chat by " + responsibleForAction.getName() + " in course (" + groupChat.getCourse().getTitle() + ").";
                return new SingleUserNotification(user, title, text);
            }
            case CONVERSATION_DELETE_CHANNEL -> {
                var channel = (Channel) conversation;
                String text = channel.getName() + " channel has been deleted by " + responsibleForAction.getName() + " in course (" + channel.getCourse().getTitle() + ").";
                return new SingleUserNotification(user, title, text);
            }
            default -> throw new UnsupportedOperationException("Unsupported NotificationType: " + notificationType);
        };
        notification.setTransientAndStringTarget(createConversationDeletionTarget(conversation, conversation.getCourse().getId()));
        notification.setAuthor(responsibleForAction);
        
        return notification;
    }
```

Reply:
i still can’t do this as setTransientAndStringTarget  depends on the notification type weather it should be `createConversationDeletionTarget ` or `createConversationCreationTarget `